### ISSUE TITLE
REFPROP Excel Add-In with Shared Files.

### ISSUE THREAD

**Issue by FloatingSam** [usnistgov/REFPROP-issues#274]
If you have an issue to report, please continue and fill out the applicable sections below.  The details provided will help to resolve this issue as quickly as possible.

### Prerequisites

**NOTE**: Be aware that all issues are publicly accessible and viewable.  Thus please do not post any code or other content that is protected intellectual property or under copyright.

Before posting an issue, please:  
    * Check the [FAQ](https://pages.nist.gov/REFPROP-docs/) for common solutions.  
    * Check that your issue isn't already filed under [REFPROP Issues](https://github.com/usnistgov/REFPROP-issues/).  
    * Check the [documentation](http://refprop-docs.readthedocs.io/en/latest/).  

You can also use markdown to format your issue: [GitHub guide on Markdown](https://guides.github.com/features/mastering-markdown/). If you have code snippets, please use a code block to ensure that the formatting remains legible in the web interface.  For instance, surround your code in triple backticks:

``` python
print('Hello world!')
```

### Description

Several of my users are using the REFPROP Excel Add-In for the formulas it offers. We're running into issue with shared files where cells will show “NAME" Because a different user is trying to access the file and it's still pointing to the file path for the add-in of the original user. How do we share files across our company and still use the add-in?

### Steps to Reproduce

1. Upload or create a file in SharePoint (our platform.)
2. Use the REFPROP Excel Add-In to reference a formula.
3. A different user accesses the file on SharePoint. The answer is initially cached but after a very short delay the field updates and the cell will show "NAME" and it references the user folder of whoever initially referenced the formula.

**Expected behavior:** [What you expect to happen]

**Actual behavior:** [What actually happens]

### Versions

**REFPROP Version:** [Product Version 8.0]
**Operating System and Version:** [OS you are running REFPROP on]  
**Access Method:** [How you are accessing REFPROP (i.e. Python, other wrapper)]  

### Additional Information

If possible, please post examples and/or screenshots of the issue.

**Comment by ianhbell** [comment]
Can you try with a more recent version of REFPROP?  We don't actively support REFPROP 8 anymore.

More generally, REFPROP is neither threadsafe nor is is process-safe for reading fluid files concurrently.  Suspect that is what you are running into.

**Comment by EricLemmon** [comment]
Indeed, the changes from version 8 to version 10 have been quite substantial!  This is especially true for mixtures.  PT diagrams for multi-component mixtures drawn with version 8 would take about 5 minutes and were very incomplete in the critical region.  Version 10 can draw these in seconds, including all calculations in the critical region.  The PT diagram for the "Gulf Coast Gas" shown below took less than three seconds.  This is a 10 component natural gas mixture.

We are currently working on a self-contained DLL that does not require any fluid files (*.FLD), the mixture file for the interaction parameters (HMX.BNC), or the *.MIX files.  If you do decide to update to version 10, let us know and we can send the DLL to you to see if it works in your environment (eric.lemmon@nist.gov).

![image](https://user-images.githubusercontent.com/42392984/77950721-28910c00-7286-11ea-9520-6e0b04f0409e.png)

**Comment by FloatingSam** [comment]
We aren't reading Fluid Files concurrently. I don't have any issue with the REFPROP Add-In Working. I have a problem with the sharing aspect. I've attached an example of what I mean. This Excel file lives in the cloud. A different user referenced a formula and now it's pointing to his file path in his "C:\Users..."location. 

How do you get around this in any version of REFPROP ?

![ProofPudding](https://user-images.githubusercontent.com/62899983/77959537-9429a800-728b-11ea-81e2-7ef37b00a043.png)

**Comment by ianhbell** [comment]
I'm not familiar with the ins and out of how sharing works with SharePoint.  Are there multiple processes running, one for user, or something like that?  If so, maybe the excel add in is locked globally?  

If you could use Python instead of Excel for analysis, then I can't see why you would have this problem in that environment.

**Comment by EricLemmon** [comment]
I believe there are some posts in this github website that address this issue, but for now let me add what I've done over the last 15 years or so.  I simply edit one of the cells, mark the part that points to the other user's XLA file, copy it, exit the cell, press Ctrl-H to do a global search and replace, paste that in, and replace every instance of it with nothing.  Then a quick Ctrl-Shift-F9 recalculates it all and you are ready to go.  I suspect you could even write a macro to do something like this where it actually finds that text by looking through the cells and, once one is found, then replaces them all.  Perhaps I'll try my luck at it one of these days.  

Now to go searching through other entries here.
